pyomo/core/expr/expr_pyomo5.py

```diff
@@ -0,0 +1,3434 @@
+'mutable_sum_context',
+'mutable_linear_context',
+'decompose_term',
+'clone_counter',
+'clone_counter_context',
+'clone_expression',
+'evaluate_expression',
+'identify_components',
+'identify_variables',
+'identify_mutable_parameters',
+'expression_to_string',
+'ExpressionBase',
+'EqualityExpression',
+'RangedExpression',
+'InequalityExpression',
+'ProductExpression',
+'MonomialTermExpression',
+'PowExpression',
+'ExternalFunctionExpression',
+'GetItemExpression',
+'Expr_ifExpression',
+'LinearExpression',
+'ReciprocalExpression',
+'NegationExpression',
+'SumExpression',
+'UnaryFunctionExpression',
+'AbsExpression',
+'compress_expression',
+'NPV_NegationExpression',
+'NPV_ExternalFunctionExpression',
+'NPV_PowExpression',
+'NPV_ProductExpression',
+'NPV_ReciprocalExpression',
+'NPV_SumExpression',
+'NPV_UnaryFunctionExpression',
+'NPV_AbsExpression',
+'SimpleExpressionVisitor',
+'ExpressionValueVisitor',
+'ExpressionReplacementVisitor',
+'LinearDecompositionError',
+'SumExpressionBase',
+'_MutableSumExpression',    # This should not be referenced, except perhaps while testing code
+'_MutableLinearExpression',     # This should not be referenced, except perhaps while testing code
+'_decompose_linear_terms',      # This should not be referenced, except perhaps while testing code
+'_chainedInequality',           # This should not be referenced, except perhaps while testing code
+'_using_chained_inequality',           # This should not be referenced, except perhaps while testing code
+'_generate_sum_expression',                 # Only used within pyomo.core.expr
+'_generate_mul_expression',                 # Only used within pyomo.core.expr
+'_generate_other_expression',               # Only used within pyomo.core.expr
+'_generate_intrinsic_function_expression',  # Only used within pyomo.core.expr
+'_generate_relational_expression',          # Only used within pyomo.core.expr
+)
+
+import math
+import logging
+import sys
+import traceback
+from copy import deepcopy
+from collections import deque
+from itertools import islice
+from six import next, string_types, itervalues
+from six.moves import xrange, builtins
+from weakref import ref
+
+logger = logging.getLogger('pyomo.core')
+
+from pyutilib.misc.visitor import SimpleVisitor, ValueVisitor
+from pyutilib.math.util import isclose
+
+from pyomo.util.deprecation import deprecation_warning
+from pyomo.core.expr.symbol_map import SymbolMap
+from pyomo.core.expr.numvalue import \
+    (NumericValue,
+     NumericConstant,
+     native_types,
+     nonpyomo_leaf_types,
+     native_numeric_types,
+     as_numeric,
+     value)
+from pyomo.core.expr.expr_common import \
+    (_add, _sub, _mul, _div,
+     _pow, _neg, _abs, _inplace,
+     _unary, _radd, _rsub, _rmul,
+     _rdiv, _rpow, _iadd, _isub,
+     _imul, _idiv, _ipow, _lt, _le,
+     _eq)
+from pyomo.core.expr import expr_common as common
+
+
+if _using_chained_inequality:               #pragma: no cover
+    class _chainedInequality(object):
```

Comment:
Why is this class definition in an if block?  That seems wrong.

Reply:
This is deprecated functionality.  Hence, this captures the logic that will be deleted.